### ISSUE TITLE
SA1515 falsely reported when xml comment immediately follows region directive.

### ISSUE THREAD

**Issue by EdwinEngelen** [DotNetAnalyzers/StyleCopAnalyzers#1371]
The code below reports SA1515 (SingleLineCommentMustBePrecededByBlankLine).
It should not report SA1515, because it is not a single line comment, but an xml comment.

```
#region TestRegion
/// <summary>
/// Test summary.
/// </summary>
public void TestMethod()
{
}
#endregion
```

**Comment by sharwell** [comment]
When XML documentation comments are disabled (SA1652), three slashes is a single line comment according to the compiler. Can you check this setting and let me know if it resolves the problem?

**Comment by EdwinEngelen** [comment]
I've changed the project setting 'XML documentation file' to 'checked' and that solved the problem.
This information should be added to the documentation.

**Comment by sharwell** [comment]
> Instead of SA1515 the compiler then reports CS1591 (Missing XML comment for publicly visible type or member) on other elements.

There are two excellent ways to handle this. You can choose the option that works best for your environment.

1. Option 1: Use the `<NoWarn>` property in your project file, as documented in the [**How to fix violations** section for SA1652](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1652.md).
2. Use a rule set file to change the severity of CS1591 and CS1573 to Hidden instead of Warning, as shown in commit 35e778f0958f321c0212a7c2571036fcdb59407d. This strategy allows the code fix for CS1591 to continue working as an on-demand refactoring operation without causing this message to show up in a warning for your builds, but only suppresses the warning when building with Visual Studio 2015.
